src/frontend/src/optimizer/plan_node/logical_over_window.rs

```diff
@@ -53,7 +53,41 @@ impl LogicalOverWindow {
         input: PlanRef,
         mut select_exprs: Vec<ExprImpl>,
     ) -> Result<(PlanRef, Vec<ExprImpl>)> {
-        let input_len = input.schema().len();
+        let mut input_proj_builder = ProjectBuilder::default();
+        let _: Vec<_> = input
+            .schema()
+            .fields()
+            .iter()
+            .enumerate()
+            .map(|(idx, x)| input_proj_builder.add_expr(&InputRef::new(idx, x.data_type()).into()))
+            .try_collect()
+            .map_err(|err| ErrorCode::NotImplemented(format!("{err} inside input"), None.into()))?;
```

Comment:
```suggestion
        for (idx, field) in input.schema().fields().iter().enumerate() {
            input_proj_builder
                .add_expr(&InputRef::new(idx, field.data_type()).into())
                .map_err(|err| {
                    ErrorCode::NotImplemented(format!("{err} inside input"), None.into())
                })?;
        }
```

Personally I think for loop can be more concise here.